File: app/src/main/java/com/tomatogaming/mealmelt/MainActivity.java
package com.tomatogaming.mealmelt;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.tomatogaming.mealmelt.Model.Recipe;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //https://www.binpress.com/android-recyclerview-cardview-guide/
        RecyclerView recList = findViewById(R.id.recyclerView);
        recList.setHasFixedSize(true);
        LinearLayoutManager llm = new LinearLayoutManager(this);
        llm.setOrientation(LinearLayoutManager.VERTICAL);
        recList.setLayoutManager(llm);

        List<Recipe> recipes = new ArrayList<>();
        Recipe testRecipe1 = new Recipe();
        testRecipe1.Name = "Test Recipe 1";
        testRecipe1.PhotoId = R.drawable.chefs_hat;
        recipes.add(testRecipe1);
        Recipe testRecipe2 = new Recipe();
        testRecipe2.Name = "Test Recipe 2";
        testRecipe2.PhotoId = R.drawable.chefs_hat;
        recipes.add(testRecipe2);
        Recipe testRecipe3 = new Recipe();
        testRecipe3.Name = "Test Recipe 3";
        testRecipe3.PhotoId = R.drawable.chefs_hat;
        recipes.add(testRecipe3);
        Recipe testRecipe4 = new Recipe();
        testRecipe4.Name = "Test Recipe 4";
        testRecipe4.PhotoId = R.drawable.chefs_hat;
        recipes.add(testRecipe4);
        Recipe testRecipe5 = new Recipe();
        testRecipe5.Name = "Test Recipe 5";
        testRecipe5.PhotoId = R.drawable.chefs_hat;
        recipes.add(testRecipe5);
        Recipe testRecipe6 = new Recipe();
        testRecipe6.Name = "Test Recipe 6";
        testRecipe6.PhotoId = R.drawable.chefs_hat;
        recipes.add(testRecipe6);
        Recipe testRecipe7 = new Recipe();
        testRecipe7.Name = "Test Recipe 7";
        testRecipe7.PhotoId = R.drawable.chefs_hat;
        recipes.add(testRecipe7);
        //TODO: populate recipes from somewhere

        RecipeAdapter recipeAdapter = new RecipeAdapter(recipes);
        recList.setAdapter(recipeAdapter);
    }
}

File: MealMelt/Activities/MainActivity.cs
using Android.App;
using Android.Content;
using Android.OS;
using Android.Support.Design.Widget;
using Android.Support.V7.App;
using Android.Support.V7.Widget;
using Android.Views;
using Java.Interop;
using MealMelt.Repository;
using System.Linq;
using System.IO;
using System;
using Android.Widget;

namespace MealMelt
{
    [Activity(Label = "@string/app_name", Theme = "@style/AppTheme", MainLauncher = true)]
    public class MainActivity : AppCompatActivity
    {
        private readonly DatabaseContext _dbContext;

        public MainActivity()
        {
            var dbPath = Path.Combine(System.Environment.GetFolderPath(System.Environment.SpecialFolder.Personal), "MealMelt.db");
            _dbContext = new DatabaseContext(dbPath);
        }

        protected override void OnResume()
        {
            FetchRecipes();
            base.OnResume();
        }

        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            SetContentView(Resource.Layout.activity_main);

            var newRecipe = FindViewById<FloatingActionButton>(Resource.Id.newRecipeButton);
            newRecipe.Click += (sender, e) =>
            {
                var intent = new Intent(this, typeof(RecipeActivity));
                StartActivity(intent);
            };
        }

        private void FetchRecipes()
        {
            var recList = FindViewById<RecyclerView>(Resource.Id.recyclerView);
            var layoutManager = new GridLayoutManager(this, 2);
            recList.SetLayoutManager(layoutManager);

            var recipes = _dbContext.Recipes.ToArray();
            var recipeAdapter = new RecipeAdapter(this, recipes);
            recList.SetAdapter(recipeAdapter);
        }

        [Export("ViewExistingRecipe")]
        public void ViewExistingRecipe(View view)
        {
            var recipeId = int.Parse(view.FindViewById<TextView>(Resource.Id.txtRecipeId).Text);
            var intent = new Intent(this, typeof(RecipeActivity));
            intent.PutExtra("RecipeId", recipeId);
            StartActivity(intent);
        }
    }
}